### PR TITLE
restrict key schema evolutions

### PR DESCRIPTION
**Pull request by lewisjkl**

### CODE REVIEW

File: avro/src/main/scala/hydra/avro/registry/SchemaRegistry.scala [hunk 1]
```diff
@@ -111,13 +111,28 @@ object SchemaRegistry {
   private def getFromSchemaRegistryClient[F[_]: Sync](schemaRegistryClient: SchemaRegistryClient): SchemaRegistry[F] =
     new SchemaRegistry[F] {
 
+      private implicit class CheckKeySchemaEvolution(schemasF: F[List[Schema]]) {
+        def checkKeyEvolution(subject: String, newSchema: Schema): F[List[Schema]] = schemasF.flatTap[Unit] {
+          case _ if !subject.endsWith("-key") => Sync[F].unit
```
Comment (bretrbs) [inline avro/src/main/scala/hydra/avro/registry/SchemaRegistry.scala:116]: Feels like a double negative. Can we do if subject.endsWith("-value") instead?

File: ingestors/kafka/src/main/scala/hydra/kafka/programs/CreateTopicProgram.scala [hunk 2]
```diff
@@ -126,3 +126,7 @@ final class CreateTopicProgram[F[_]: Bracket[*[_], Throwable]: Sleep: Logger](
     } yield ()).use(_ => Bracket[F, Throwable].unit)
   }
 }
+
+object CreateTopicProgram {
```
Comment (bretrbs) [inline ingestors/kafka/src/main/scala/hydra/kafka/programs/CreateTopicProgram.scala:130]: Is there something that was supposed to be placed here that wasn't or can we delete this companion object?

File: avro/src/main/scala/hydra/avro/registry/SchemaRegistry.scala (hunk 1 above)
Comment (lewisjkl) [reply avro/src/main/scala/hydra/avro/registry/SchemaRegistry.scala:116]: I was mainly thinking that there _could_ be cases when we register subjects that don't end in `-key` or `-value`. But I guess without having a concrete use case for that you have a valid point. I will change it 🙂

File: ingestors/kafka/src/main/scala/hydra/kafka/programs/CreateTopicProgram.scala (hunk 2 above)
Comment (lewisjkl) [reply ingestors/kafka/src/main/scala/hydra/kafka/programs/CreateTopicProgram.scala:130]: Thanks, I was going to put an error there, but then decided to go with the algebra instead. Will remove.

**Pull request merged**